compiler/src/org.graalvm.compiler.core/src/org/graalvm/compiler/core/gen/NodeLIRBuilder.java

```diff
@@ -387,6 +387,8 @@ public void doBlock(Block block, StructuredGraph graph, BlockMap<List<Node>> blo
                         debug.log("interior match for %s", valueNode);
                     } else if (operand instanceof ComplexMatchValue) {
                         debug.log("complex match for %s", valueNode);
+                        // Set current position to the position of the root matched node.
+                        setSourcePosition(node.getNodeSourcePosition());
```

Comment:
Please move this up to `if (node instanceof ValueNode) {` and remove the call in `emitNode` so it is used commonly for all cases.

Reply:
Thanks for your suggestion! But `emitNode ` will not be called when "if (operand instanceof ComplexMatchValue)" is tue, and it is just called in `doRoot`, which is in the general branch. Also I have checked that there is already a "setSourcePosition(node.getNodeSourcePosition());" calling in `emitNode `. So is there any other better place to remove the function? Thanks!

Reply:
Sorry, I just misunderstand your advise! So I will move this function call to `if (node instanceof ValueNode) {` and remove the call in `emitNode`. Thank you!